### ISSUE TITLE
DrumKit OS Loader with Pitchbend

### ISSUE THREAD

**Issue by tkna91** [esaruoho/org.lackluster.Paketti.xrnx#37]
From #34

1. Opens a Dialog
2. Lets you select multiple samples
3. Press OK
4. the Pitchbend Instrument is loaded
5. Enable Keyzones Drum Kit
6. Load all selected samples in one instrument

![swappy-20240428_185951](https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/102382754/52fed6de-419f-4cf4-901e-0f813c2557c4)

**Comment by esaruoho** [comment]
thanks, sounds interesting. let's see if drumkit is mentioned in the API

ok looks like there is no way of turning drumkit on or off. what i'm trying now is starting with yet another pre-saved instrument, in this case, 12st_pitchbend_drumkit.xrni ( 
[12st_Pitchbend_Drumkit.xrni.zip](https://github.com/esaruoho/org.lackluster.Paketti.xrnx/files/15142323/12st_Pitchbend_Drumkit.xrni.zip)

fingers crossed this'll work. i'm not sure.
anyway this requires a bunch of work before i can close this, but thanks for the ideas anyway.

**Comment by esaruoho** [comment]
some more musings about single ("load samples into single instrument instead of multiple")
```
the single instrument loader would be "selection to drumkit" and the multiple instrument loader would be 1 sample per 1 new instrument. once i get the drumkit loader working, i can rename them like this and differentiate.
```

**Comment by esaruoho** [comment]
@tkna91 hi, i think i'm gonna need to work on this next since i have 40+ tickets open now.

could you look through them and weigh in on some of them and leave some more ideas?

**Comment by tkna91** [comment]
That's a great amount of activity.
In haste, this is the only thing that caught my attention.
https://github.com/esaruoho/org.lackluster.Paketti.xrnx/issues/46
Sorry I can't help much as I haven't been very active lately, but I'm rooting for you.

**Comment by esaruoho** [comment]
@tkna91 https://www.loom.com/share/d74b32aea5f942cc9c4315a1307b8406
so i got something like this working. it's called
<img width="348" alt="Screenshot 2024-05-22 at 12 34 27" src="https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/4966687/398ffc7b-a344-4195-9c6f-ff9baf63ef0c">

and if you want to take it for a spin, here it is.

please let me know any input. would you prefer to start from C1 instead of C3 for instance.

any videos and any feedback would be appreciated.

and i apologize for taking so long to do this, i was doing really low-hanging-fruit type features and got sidetracked.

btw are you on discord? please send me a message to esaruoho, if you are.

[org.lackluster.Paketti_V3.54.xrnx.zip](https://github.com/esaruoho/org.lackluster.Paketti.xrnx/files/15401267/org.lackluster.Paketti_V3.54.xrnx.zip)

**Comment by esaruoho** [comment]
hmm looking at your screenshot, you'd prefer to start from C0. so i'll do that then.
<img width="843" alt="Screenshot 2024-05-22 at 12 40 48" src="https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/4966687/d66658c5-1343-471f-b4ca-6816a52e5e17">

ok i gotta limit the loading so that it fills C0 to B9. i'll need to calculate how many sample slots that is.
[org.lackluster.Paketti_V3.54.xrnx.zip](https://github.com/esaruoho/org.lackluster.Paketti.xrnx/files/15401423/org.lackluster.Paketti_V3.54.xrnx.zip)

so it'll load 119 samples.
C0 will be quiet, until i tweak that so the first sample is loaded there. i need to have a placeholder, set to "Drumkit" mode, so that it will definitely populate the loaded samples as drums. and if you pick something like 800 samples, then:
<img width="517" alt="Screenshot 2024-05-22 at 12 53 32" src="https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/4966687/f4afd570-a5d2-47e6-958c-7930569d5c2c">

@tkna91 please check this instead of the other.

**Comment by esaruoho** [comment]
and here's a video of it in action
https://www.loom.com/share/222d40052951408bb65cdf76a746ceaf?sid=1aff5e91-d506-4baf-a2f7-e7287b7c21dd

**Comment by tkna91** [comment]
I think it would be better to read the start of the sample from `from` instead of fixing it at C0.
![image](https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/102382754/2e7ebaba-fb1a-476d-a3c8-a7bd7875c56a)

It seems to start with an empty sample `placeholder for drumkit`.
Is this necessary?
![image](https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/102382754/9e9ebdbe-b9d7-49d6-a89d-4e6bdff3c07e)

**Comment by tkna91** [comment]
if so, I think ok.
thanks!

**Comment by esaruoho** [comment]
i unfortunately found a bug with this. if there's already a `*Instr. Automation` in the track, it will mess up. will tweak and then close after this is no longer an issue.

**Comment by esaruoho** [comment]
ok, it was an older .xrni file.
so i'm adding some improvements.

1. create a new instrument before starting the process
2. if nothing was selected (selection was NIL, exit the script)
3. 120 samples are loaded

i'll eventually need to make it so that *Instr Macros is named but that's a separate ticket at #58

**Comment by esaruoho** [comment]
<img width="1212" alt="Screenshot 2024-05-30 at 10 46 52" src="https://github.com/esaruoho/org.lackluster.Paketti.xrnx/assets/4966687/e506791f-337d-42d7-8624-76dec5270366">

new instrument, exit script, 120 samples loaded, and instr macro naming works as expected.

closing.
